src/peerstate.rs

```diff
@@ -164,6 +165,22 @@ impl Peerstate {
         Self::from_stmt(context, query, paramsv![fp, fp, fp]).await
     }
 
+    pub async fn from_nongossiped_fingerprint_or_addr(
+        context: &Context,
+        fingerprint: &Fingerprint,
+        addr: &str,
+    ) -> Result<Option<Peerstate>> {
+        let query = "SELECT addr, last_seen, last_seen_autocrypt, prefer_encrypted, public_key, \
+                     gossip_timestamp, gossip_key, public_key_fingerprint, gossip_key_fingerprint, \
+                     verified_key, verified_key_fingerprint \
+                     FROM acpeerstates  \
+                     WHERE public_key_fingerprint=? COLLATE NOCASE \
+                     OR addr=? COLLATE NOCASE \
+                     ORDER BY public_key_fingerprint=? DESC, last_seen DESC;";
```

Comment:
@link2xt I just noticed that in this line (and in `from_fingerprint()` above) there is probably a COLLATE NOCASE missing?

`ORDER BY public_key_fingerprint=? DESC COLLATE NOCASE, last_seen DESC;`

Also, I was wondering whether this may be sped up with `LIMIT 1`, what does your SQL intuition say about this?

Reply:
Then we can probably remove the NOCASE in `public_key_fingerprint=? COLLATE NOCASE`, too?

Reply:
Also, I just re-checked, both in the code and in my database, they are indeed all uppercase(d)